crates/subspace-farmer/src/bin/subspace-farmer/commands/farm.rs

```diff
@@ -238,14 +238,14 @@ fn networking_getter(plots: &[Plot], key: &Multihash) -> Option<Vec<u8>> {
 
     let piece_index = u64::from_le_bytes(key.digest()[..mem::size_of::<u64>()].try_into().ok()?);
 
-    let (mut piece, public_key) = plots.iter().find_map(|plot| {
-        plot.read_piece(piece_index)
-            .ok()
-            .map(|piece| (piece, plot.address()))
-    })?;
+    let plot = plots
+        .iter()
+        .min_by_key(|plot| PieceDistance::xor_distance(&piece_index.into(), plot.address()))
+        .expect("We always have at least one plot");
```

Comment:
Actually 2 closest need to be checked. Otherwise the very last small plot might be closest, but not actually having the piece due to its size.